pkg/controller/gcp/compute/network.go

```diff
@@ -0,0 +1,180 @@
+type NetworkController struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *NetworkController) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(v1alpha1.NetworkGroupVersionKind),
+		resource.WithExternalConnecter(&networkConnector{kube: mgr.GetClient()}),
+		resource.WithManagedConnectionPublishers())
+
+	name := strings.ToLower(fmt.Sprintf("%s.%s", v1alpha1.NetworkKindAPIVersion, v1alpha1.Group))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Network{}).
+		Complete(r)
+}
+
+type networkConnector struct {
+	kube         client.Client
+	newServiceFn func(ctx context.Context, opts ...option.ClientOption) (*googlecompute.Service, error)
+}
+
+func (c *networkConnector) Connect(ctx context.Context, mg resource.Managed) (resource.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.Network)
+	if !ok {
+		return nil, errors.New(errNotNetwork)
+	}
+	// TODO(muvaf): we do not yet have a way for configure the Spec with defaults for statically provisioned resources
+	// such as this. Setting it directly here does not work since managed reconciler issues updates only to
+	// `status` subresource. We require name to be given until we have a pre-process hook like configurator in Claim
+	// reconciler
+	if cr.Spec.Name == "" {
+		return nil, errors.New(errNameNotGiven)
+	}
+
+	provider := &gcpapis.Provider{}
+	n := meta.NamespacedNameOf(cr.Spec.ProviderReference)
+	if err := c.kube.Get(ctx, n, provider); err != nil {
+		return nil, errors.Wrapf(err, "cannot get provider %s", n)
+	}
+
+	gcpCreds, err := gcpclients.ProviderCredentials(c.kube, provider, googlecompute.ComputeScope)
+	if err != nil {
+		return nil, err
+	}
+	if c.newServiceFn == nil {
+		c.newServiceFn = googlecompute.NewService
+	}
+	s, err := c.newServiceFn(ctx, option.WithCredentials(gcpCreds))
+	if err != nil {
+		return nil, errors.Wrap(err, errNewClient)
+	}
+	return &networkExternal{Service: s, projectID: provider.Spec.ProjectID}, nil
+}
+
+type networkExternal struct {
+	*googlecompute.Service
+	projectID string
+}
+
+func (c *networkExternal) Observe(ctx context.Context, mg resource.Managed) (resource.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Network)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errNotNetwork)
+	}
+	observed, err := c.Networks.Get(c.projectID, cr.Spec.Name).Context(ctx).Do()
+	if gcpclients.IsErrorNotFound(err) {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+	if err != nil {
+		return resource.ExternalObservation{}, err
+	}
+	cr.Status.GCPNetworkStatus = network.GenerateGCPNetworkStatus(*observed)
+	// If the Network resource is retrieved, it is ready to be used
+	cr.Status.SetConditions(runtimev1alpha1.Available())
```

Comment:
I believe you will want to call `cr.SetConditions()` as opposed to `cr.Status.SetConditions()`

Reply:
They're actually equivalent; `cr.SetConditions()` is just a wrapper for `cr.Status.SetConditions()`. It exists to satisfy the `managed.Resource` interface, where we want to be able to set conditions without knowing the structure of the concrete type.